### PR TITLE
scripts: linux: add setup script for linux

### PR DESCRIPTION
**Pull request by DanielGuramulta**
Addresses #82 / script for linux

Signed-off-by: Daniel Guramulta <daniel.guramulta@analog.com>

### CODE REVIEW

File: scripts/linux/setup.sh [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        esac
+        done
+        set -- "${POSITIONAL[@]}" # restore positional parameters
+
+        if [[ "${display_help}" == "True" ]]; then
+                print_help
+                exit
+        fi
+        insta
+        ll_required_packages
```
Comment (dNechita) [inline scripts/linux/setup.sh]: this seems broken in 2 lines

File: scripts/linux/setup.sh [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+        get_deps_source_code ${deps_dir}
+
+        build_and_install_glog ${deps_dir}/glog ${deps_install_dir}/glog
+        build_and_install_protobuf ${deps_dir}/protobuf ${deps_install_dir}/protobuf
+        build_and_install_websockets ${deps_dir}/libwebsockets ${deps_install_dir}/websockets
+
+        pushd "${build_dir}"
+        cmake "${source_dir}" -DWITH_PYTHON=on -DWITH_OPENCV=on -DCMAKE_PREFIX_PATH="${deps_install_dir}/glog;${deps_install_dir}/protobuf;${deps_install_dir}/websockets"
+        make -j
```
Comment (dNechita) [inline scripts/linux/setup.sh]: ${NUM_JOBS}?

File: scripts/linux/setup.sh (hunk 1 above)
Comment (DanielGuramulta) [reply scripts/linux/setup.sh]: Something must have happened when I formatted the file, Thank you!

File: scripts/linux/setup.sh (hunk 2 above)
Comment (DanielGuramulta) [reply scripts/linux/setup.sh]: I couldn't find a way to get the number of cores for the system. I think if it is not to much work an option like `-c x` for example that will set `NUM_JOBS=x` will work, or even using just `make` as it is right now will work. Indeed I forgot to remove the `-j`. Thanks.

Comment (dNechita) [reply scripts/linux/setup.sh]: You can use `nproc --all` or use the -c option or both (use nproc by default and overwrite if user provides the core count)

Comment (DanielGuramulta) [reply scripts/linux/setup.sh]: I have added a `-j|--jobs` option, as I see `-j` is used more often to specify the number of concurrent jobs. If that is not specified, the number of jobs will be the return value of `nproc --all`. Thanks for your suggestion.

**Pull request merged**